### PR TITLE
Add requests.Session compatibility docs

### LINKED ISSUE #403: More explicit documentation that `httpx.Client` is equivelent to `requests.Session`.
This isn't an issue (sorry if i posted in a wrong place), but i read the httpx doc and didn't find yet a solution. 
In requests i use s = requests.session(), how can i do the same thing with httpx? 
it will be possible to do something like s = httpx.session()?
Thanks and sorry again.

### PR DESCRIPTION
**Pull request by jtmiclat**
Resolves #403

### CODE REVIEW

**Comment by tomchristie** [comment]
Grand. I don’t think we particularly need to mention AsyncClient here, since requests doesn’t have any equivalent.

**Comment by jtmiclat** [comment]
@tomchristie Update the docs to remove AsyncClient!

File: docs/compatibility.md [hunk 1]
```diff
@@ -17,5 +17,14 @@ but also provide lower-cased versions for API compatibility with `requests`.
...
+HTTPX equivalent of `requests.Session` is `httpx.Client` for standard synchronous API.
+
+```python
+session = requests.Session(**kwargs)
+```
+is equivalent to
+```python
+client = httpx.Client(**kwargs)
+```
+More detailed documentation and usage of `Client` can be found in [Advanced Usage](advanced.md)
```
Comment (florimondmanca) [inline docs/compatibility.md]:
```suggestion
More detailed documentation and usage of `Client` can be found in [Advanced Usage](advanced.md).
```

File: docs/compatibility.md [hunk 2]
```diff
@@ -17,5 +17,14 @@ but also provide lower-cased versions for API compatibility with `requests`.
 
 ## Advanced Usage
 
-!!! warning
-    TODO
+### requests.Session
+HTTPX equivalent of `requests.Session` is `httpx.Client` for standard synchronous API.
```
Comment (florimondmanca) [inline docs/compatibility.md]:
```suggestion
The HTTPX equivalent of `requests.Session` is `httpx.Client`.
```

File: docs/advanced.md [hunk 3]
```diff
@@ -6,6 +6,8 @@ Using a Client instance to make requests will give you HTTP connection pooling,
 will provide cookie persistence, and allows you to apply configuration across
 all outgoing requests.
 
+A Client instance is equivalent to a Session instance of `requests`.
```
Comment (florimondmanca) [inline docs/advanced.md]:
```suggestion
A Client instance is equivalent to a Session instance in `requests`.
```

File: docs/compatibility.md [hunk 4]
```diff
@@ -17,5 +17,14 @@ but also provide lower-cased versions for API compatibility with `requests`.
...
+The HTTPX equivalent of `requests.Session` is `httpx.Client`.
+
+```python
+session = requests.Session(**kwargs)
+```
+is equivalent to
+```python
+client = httpx.Client(**kwargs)
+```
+More detailed documentation and usage of `Client` can be found in [Advanced Usage](advanced.md).
```
Comment (florimondmanca) [inline docs/compatibility.md]: Sorry I didn't get this in my first review — should we try to keep the Markdown formatting consistent here? I believe it is encouraged to put lines between various elements, i.e.:

```markdown
### requests.Session

The HTTPX equivalent of `requests.Session` is `httpx.Client`.

\```python
session = requests.Session(**kwargs)
\```

is equivalent to:

\```python
client = httpx.Client(**kwargs)
\```

More detailed documentation and usage of `Client` can be found in [Advanced Usage](advanced.md).
```

**Pull request merged**